transport/handler.go

```diff
@@ -37,3 +37,17 @@ type Handler interface {
 		resw ResponseWriter,
 	) error
 }
+
+// OnewayHandler handles a single, transport-level, oneway request.
+type OnewayHandler interface {
+	// Handle the given oneway request
+	//
+	// An error may be returned in case of failures. BadRequestError must be
+	// returned for invalid requests. All other failures are treated as
+	// UnexpectedErrors.
+	HandleOneway(
+		ctx context.Context,
+		opts Options,
+		req *Request,
+	) error
```

Comment:
TBD right now regarding the goroutine. We plan on implementing a OnewayError so that the server can decide whether to retry or not for example.